src/main/java/htmlpublisher/HtmlPublisherTarget.java

```diff
@@ -179,18 +181,27 @@ protected File dir() {
         protected String getTitle() {
             return this.project.getDisplayName() + " html2";
         }
+        
+        /**
+         * Gets {@link HtmlPublisherTarget}, for which the action has been created.
+         * @return HTML Report description
+         * @since TODO
+         */
+        public @Nonnull HtmlPublisherTarget getHTMLTarget() {
+            return HtmlPublisherTarget.this;
+        }
     }
 
-    public class HTMLBuildAction extends BaseHTMLAction {
-        private final AbstractBuild<?, ?> build;
+    public class HTMLBuildAction extends BaseHTMLAction implements RunAction2 {
+        private transient Run<?, ?> build;
 
-        public HTMLBuildAction(AbstractBuild<?, ?> build, HtmlPublisherTarget actualHtmlPublisherTarget) {
+        public HTMLBuildAction(Run<?, ?> build, HtmlPublisherTarget actualHtmlPublisherTarget) {
```

Comment:
The interface is public. It makes sense to deprecate the method, but we still need to maintain the backward compatibility